### DISCUSSION TITLE
Chart SQL based on

### DISCUSSION THREAD

**Discussion by supermarin1** [apache/superset#14134] [Q&A / Help]
I'm interested in SQL that's used to retrieve the data for the chart. To get in through the API

Regarding to this -> https://github.com/apache/superset/issues/14097#issuecomment-819160022
I was told to use  POST `/api/v1/chart/data`with `form_data` parameter.  

This endpoint requires JSON body and it responses back with data. `query` in response is based on request's body not the same as I can get from UI (chart -> menu -> view query)

Please, any advice how can I get that SQL?

**Comment by one-data-cookie** [comment]
@supermarin1 Any chance you ever got this figured out? 🙏

**Reply by mestachs** [reply]
I'm interested in that too. 
My use case validate queries on my CI or laptop before putting a new version of my dataset.
but the post payload it much complex that what I expect, how to build it ?

**Reply by mestachs** [reply]
part of the answer, but doesn't look trivial

https://github.com/apache/superset/discussions/33574#discussioncomment-13256494

I found the equivalent code of the front end "buildQueryContext" 

https://github.com/apache/superset/blob/master/superset-frontend/packages/superset-ui-core/src/query/buildQueryContext.ts#L35

in the backend (which looks better for my usecase, since I'm in python)

https://github.com/apache/superset/blob/b24aca0304b53cdf6f92bf600b64152015cafa30/superset/migrations/shared/migrate_viz/query_functions.py#L1054-L1087
